x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/rest/action/IdpBaseRestHandlerTests.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.idp.saml.rest.action;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.TestUtils;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class IdpBaseRestHandlerTests extends ESTestCase {
+
+    public void testIdpAvailableOnTrialOrEnterprise() {
+        final IdpBaseRestHandler handler = buildHandler(randomFrom(License.OperationMode.ENTERPRISE, License.OperationMode.TRIAL));
+        assertThat(handler.isIdpFeatureAllowed(), equalTo(true));
+    }
+
+    public void testIdpNotAvailableOnOtherLicenses() {
+        final IdpBaseRestHandler handler = buildHandler(randomFrom(
+            License.OperationMode.BASIC, License.OperationMode.GOLD, License.OperationMode.MISSING, License.OperationMode.MISSING,
```

Comment:
`MISSING` is here twice.

The other option, rather than explicitly listing non-enterprise modes is to use `randomValueOtherThanMany` like:

```
License.OperationMode mode = randomValueOtherThanMany( m -> m == License.OperationMode.ENTERPRISE || m == License.OperationMode.TRIAL, () -> randomFrom(License.OperationMode.values()) )
```